packages/flutter_tools/lib/src/commands/drive.dart

```diff
@@ -42,19 +42,22 @@ class DriveCommand extends RunCommandBase {
     argParser.addFlag(
       'keep-app-running',
       negatable: true,
-      defaultsTo: false,
       help:
         'Will keep the Flutter application running when done testing.\n'
-        'By default, Flutter drive stops the application after tests are finished.\n'
-        'Ignored if --use-existing-app is specified.'
+        'By default, "flutter drive" stops the application after tests are finished,\n'
+        'and --keep-app-running overrides this. On the other hand, if --use-existing-app\n'
+        'is specified, then "flutter drive" instead defaults to leaving the application\n'
+        'running, and --no-keep-app-running overrides it.'
     );
 
     argParser.addOption(
       'use-existing-app',
       help:
         'Connect to an already running instance via the given observatory URL.\n'
-        'If this option is given, the application will not be automatically started\n'
-        'or stopped.'
+        'If this option is given, the application will not be automatically started,\n'
+        'and it will only be stopped if --no-keep-app-running is explicitly set.',
+      valueHelp:
+        'url'
```

Comment:
Yeah I filed a bug on the docs being suboptimal.

It makes the help say that the syntax is `--keep-app-running=<url>`.